### DISCUSSION TITLE
Supporting Arm models in Bochs ?

### DISCUSSION THREAD

**Discussion by stlintel** [bochs-emu/Bochs#346] [Ideas]
Hi,

I am interesting to learn Arm internals and I would like to implement Arm CPU model into Bochs.
I became x86 expert doing it for x86 and now want to get into Arm - this is my way of learning ISA, instructions sets and software.
I wish Bochs would be able to support some Arm platform like QEMU does.
How hard would it get some Arm platform emulating ? 
At least as quick start. May be port one one the boards supported in Qemu...
What do you think ?

Thanks,

**Comment by Vort** [comment]
1. Expanding of project scope comes with risks and requires careful thought;
2. If I understand correctly, x86 hardware is relatively standardized and interchangable, while Arm hardware is specific to each device. Implementing hundreds of hardware variations will require enormous amount of resources.

**Reply by stlintel** [reply]
QEMU does it and you may see how it looks like.
Arm ISA is standardized so CPU model would be same for all.
Devices (mainly mother boards) is a mess but I am not asking to have all of them.
Is it possible to pick up one device supported by QEMU (going to use Qemu as refence model for CPU for now) and port it here ?
Old Bochs code won't be at risk trivially, the cpu model will be developed in separate folder.
The new devices are new devices and if not enabled have no effect on legacy machines emulated
